iam.tf

```diff
@@ -28,3 +28,206 @@ resource "aws_iam_role_policy" "cluster_autoscaler" {
     }
   )
 }
+
+# TODO: we might want to add this policy to the service account that the load balancer controller uses
+#   instead of making it globally available to the whole worker node.
+#   (see: https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/deploy/installation/#iam-permissions)
+
+resource "aws_iam_role_policy" "aws_load_balancer_controller" {
+  count = var.install_load_balancer_controller ? 1 : 0
+
+  name = "aws-load-balancer-policy"
+  role = data.aws_iam_role.worker_role.id
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "iam:CreateServiceLinkedRole",
+          "ec2:DescribeAccountAttributes",
+          "ec2:DescribeAddresses",
+          "ec2:DescribeInternetGateways",
+          "ec2:DescribeVpcs",
+          "ec2:DescribeSubnets",
+          "ec2:DescribeSecurityGroups",
+          "ec2:DescribeInstances",
+          "ec2:DescribeNetworkInterfaces",
+          "ec2:DescribeTags",
+          "elasticloadbalancing:DescribeLoadBalancers",
+          "elasticloadbalancing:DescribeLoadBalancerAttributes",
+          "elasticloadbalancing:DescribeListeners",
+          "elasticloadbalancing:DescribeListenerCertificates",
+          "elasticloadbalancing:DescribeSSLPolicies",
+          "elasticloadbalancing:DescribeRules",
+          "elasticloadbalancing:DescribeTargetGroups",
+          "elasticloadbalancing:DescribeTargetGroupAttributes",
+          "elasticloadbalancing:DescribeTargetHealth",
+          "elasticloadbalancing:DescribeTags"
+        ],
+        "Resource" : "*"
```

Comment:
Is it possible to constrain all these `"Resource" : "*"` into something like `"Resource" : "arn:aws:<service>:<account>:*"`?

Reply:
Probably. But for all our use-cases it won't make much of a difference. Making this policy attached to a pod service account would be ideal, but that's left as a future improvement.